crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -413,6 +417,21 @@ fn handle_trailing_body_comment<'a>(
     // the indent-level doesn't depend on the tab width (the indent level must be the same if the tab width is 1 or 8).
     let comment_indentation_len = comment_indentation.len();
 
+    // Keep the comment on the if it's a trailing comment
+    // ```python
+    // if "first if":
+    //     pass
+    // elif "first elif":
+    //     pass
+    // # Trailing if comment
+    // ```
+    let Some(preceding_node_indentation) = whitespace::indentation_at_offset(locator, preceding_node.start()) else {
+        return CommentPlacement::Default(comment);
+    };
+    if comment_indentation_len == preceding_node_indentation.len() {
+        return CommentPlacement::Default(comment);
+    }
```

Comment:
This seems to me to be the same logic as inside of the loop. Could we change `parent_body` to `Option` and start with `None`, and then initialize `current_child` with `comment.preceding_node` to avoid repeating the same logic?

Reply:
I tried that, but then we start adding a lot of extraneous whitespace, i'm pretty sure we need the `Default` as a base because `Ordering::Equal` always sets it as trailing